dd-sdk-android/src/main/kotlin/com/datadog/android/sessionreplay/internal/domain/SessionReplayRequestFactory.kt

```diff
@@ -109,45 +89,33 @@ internal class SessionReplayRequestFactory(
     }
 
     @Suppress("ReturnCount")
-    private fun resolveRequest(context: DatadogContext, body: RequestBody?): Request {
-        if (body != null) {
-            val bodyAsByteArray = safeBodyToByteArray(body) ?: return getEmptyRequest()
-            val requestId = UUID.randomUUID().toString()
-            val description = "Session Replay Segment Upload Request"
-            val headers = resolveHeaders(context, requestId)
-            val requestUrl = buildUrl(context)
-            return Request(
-                requestId,
-                description,
-                requestUrl,
-                headers,
-                body = bodyAsByteArray,
-                contentType = body.contentType().toString()
-            )
-        } else {
-            return getEmptyRequest()
-        }
+    private fun resolveRequest(context: DatadogContext, body: RequestBody): Request {
+        val bodyAsByteArray = extractByteArrayFromBody(body)
+        val requestId = UUID.randomUUID().toString()
+        val description = "Session Replay Segment Upload Request"
+        val headers = resolveHeaders(context, requestId)
+        val requestUrl = buildUrl(context)
+        return Request(
+            requestId,
+            description,
+            requestUrl,
+            headers,
+            body = bodyAsByteArray,
+            contentType = body.contentType().toString()
+        )
     }
 
-    private fun safeBodyToByteArray(body: RequestBody): ByteArray? {
-        return try {
-            bodyToByteArrayFactory(body)
-        } catch (@Suppress("TooGenericExceptionCaught") e: Throwable) {
-            internalLogger.log(
-                InternalLogger.Level.ERROR,
-                InternalLogger.Target.MAINTAINER,
-                EXTRACT_BYTE_ARRAY_FROM_BODY_ERROR_MESSAGE,
-                e
-            )
-            null
-        }
+    private fun extractByteArrayFromBody(body: RequestBody): ByteArray {
+        val buffer = Buffer()
+        @Suppress("UnsafeThirdPartyFunctionCall")
+        body.writeTo(buffer)
+        @Suppress("UnsafeThirdPartyFunctionCall")
+        return buffer.readByteArray()
```

Comment:
this `readByteArray` method is strange - it says that it can throw `IOException`, but the actual implementation actually can also throw `Error`:

```
    try {
      return readByteArray(size);
    } catch (EOFException e) {
      throw new AssertionError(e);
    }
```

Do you think we should catch it? Because the outer code only handles exceptions, but not errors. Normally errors are not recoverable though, so maybe we don't need to.

Reply:
yeah...that's the thing...I would leave errors not caught ...it will be another thread anyway not the main thread.